whisperv6/peer.go

```diff
@@ -192,21 +201,35 @@ func (peer *Peer) expire() {
 	}
 }
 
+func (peer *Peer) reduceBundle(bundle []*Envelope) []*Envelope {
+	if peer.host.ratelimiter == nil {
+		return bundle
+	}
+	for i := range bundle {
+		size := int64(bundle[i].size())
+		if peer.host.ratelimiter.Egress.Available(peer.peer) < size {
+			return bundle[:i]
+		}
+		peer.host.ratelimiter.Egress.TakeAvailable(peer.peer, size)
+	}
+	return bundle
+}
+
 // broadcast iterates over the collection of envelopes and transmits yet unknown
 // ones over the network.
 func (peer *Peer) broadcast() error {
 	if peer.peer.IsFlaky() {
 		log.Trace("Waiting for a peer to restore communication", "ID", peer.peer.ID())
 		return nil
 	}
-	envelopes := peer.host.Envelopes()
+	envelopes := peer.host.Envelopes() // envelopes are read from hash map, so access is already randomized
 	bundle := make([]*Envelope, 0, len(envelopes))
 	for _, envelope := range envelopes {
 		if !peer.marked(envelope) && envelope.PoW() >= peer.powRequirement && peer.bloomMatch(envelope) {
 			bundle = append(bundle, envelope)
 		}
 	}
-
+	bundle = peer.reduceBundle(bundle)
```

Comment:
A comment or method name change would be nice. By looking at this fragment without having a context, I would have no idea why we reduce bundle.